app/scripts/directives/mobileServiceInstanceRow.js

```diff
@@ -0,0 +1,125 @@
+        DataService.unwatch(bindingsWatch);
+    });
+
+    watches.push(DataService.watchObject(APIService.getPreferredVersion('mobileclients'), $routeParams.mobileclient, {namespace: $routeParams.project}, function(mobileClient) {
+        row.mobileClient = mobileClient;
+
+        row.annotations = _.filter(row.mobileClient.metadata.annotations, function(annotation, name){
+          return name.split("/")[0] === annotationSpace;
+        })
+        .map(JSON.parse)
+        .sort(function(a, b) {
+          if (a.label < b.label) {
+            return -1;
+          }
+          if (a.label > b.label) {
+            return 1;
+          }
+          return 0;
+        });
+
+        row.serviceType = _.get(row, "serviceClass.spec.externalMetadata.serviceName"); // safe get
+
+        row.extendedAnnotations = _(row.mobileClient.metadata.annotations)
+          .pickBy(function(annotation, name){
+            return name.split("/")[0] === extendedAnnotationSpace;
+          })
+          .transform(function(result, strVal, name) {
+            var key = name.split("/")[1];
+            result[key] = JSON.parse(strVal);
+          })
+          .value();
+
+        row.servicePlanPromise = DataService.list(APIService.getPreferredVersion('clusterserviceplans'), {}, function(servicePlans) {
+          var servicePlanData = servicePlans.by('metadata.name');
+          row.providerServicePlan = _.find(servicePlanData, function(plan) {
+            return row.serviceClass.metadata.name === plan.spec.clusterServiceClassRef.name;
+          });
+          var kind = row.mobileClient.kind.toLowerCase();
+          var bindDataPath = 'providerServicePlan.spec.externalMetadata.' + kind + '_bind_parameters_data';
+          var bindData = _.get(row, bindDataPath, []);
+          row.bindData = _(bindData).map(JSON.parse).value();
+        });
+
+        if(!bindingsWatch){
+          bindingsWatch = DataService.watch(APIService.getPreferredVersion("servicebindings"), {namespace: $routeParams.project}, function(bindings){
+            row.bindings = _.filter(bindings.by("metadata.name"), function(binding){
+              return _.get(binding.metadata.annotations, "binding.aerogear.org/consumer") === row.mobileClient.metadata.name &&
+                    _.get(binding.metadata.annotations, "binding.aerogear.org/provider") === row.apiObject.metadata.name;
+            });
+
+            row.bindingInProgress = _.sumBy(row.bindings, function(binding){
+              return $filter('isBindingReady')(binding) ? 0 : 1;
+            });
+          });
+        }
+    }));
+
+    row.$onChanges = function() {
+      row.displayName = serviceInstanceDisplayName(row.apiObject, row.serviceClass);
+    };
+
+    row.closeOverlayPanel = function() {
+      _.set(row, 'overlay.panelVisible', false);
+    };
+
+    row.showOverlayPanel = function(panelName, state) {
+      row.parameterData = _.reduce(row.bindData, function(acc, current) {
```

Comment:
`parameterData` is passed in as an attribute, just want to check & verify you do need 2-way data binding here.

Reply:
No, I believe one way is correct. The child component only uses `parameterData`